### PR TITLE
B 19531 int

### PR DESCRIPTION
**Pull request by antgmann**
## Summary

This task has been repurposed since it's original creation in accordance with the fluctuation of DTOD changes. It is now a follow-on to [B-20757](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20757) (see [int pr 1](https://github.com/transcom/mymove/pull/13789), [int pr 2](https://github.com/transcom/mymove/pull/13859) ), which takes the stopgap error message 20757 and also introduces it in the Prime Simulator for updating shipments.

Testing 20757 in our integration/loadtest environment has been tricky due to the current dtod implementation in that environment. This issue will likely hold up this ticket as well, but it can be tested on local environments with a little setup.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [x] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### How to test

1. First before testing, navigate to pkg/route/dtod_zip5_distance.go and add `distanceFloat = -1` on line 124. This will simulate a dtod outage and allow us to continue testing. This is the step that makes testing tricky on other environments with their dtod setups.
2. Once this has been completed, log in as a prime simulator user and navigate to any move. I've been using move code PARAMS to test for HHGs.
3. Select "Update Shipment" for any shipment. 
4. Change any values, such as "Destination Type" at the bottom, and click update.
5. After a pause, the error should appear at the top with the correct message.
6. That's HHG's sorted. PPMs are a little different, but to test them, you must log in as a customer and make a new move.
7. Add a PPM Shipment, and set a duty location to NS Mayport. This is our location for Prime-counseled PPMs.
8. Submit the shipment with any data.
9. Log in as a TOO. Fill out orders info and approve to send the shipment to Prime.
10. Log in as a prime, find your prime-counseled PPM.
11. Update the shipment.
12. Under either zip code, you'll need to provide a new zip code to hit the DTOD section of code. If you're simulating the DTOD outage still, any zip code will do. Otherwise, 23471 is a good zip code that fails as it's a PO box. 
13. After updating the zipcode, click submit, and the error message should appear.

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

## Screenshots
image

I don't know that ANDI is required for prime sim, but either way it passes:
image

### CODE REVIEW

**Comment by antgmann** [comment]
Deleted comments to reset test coverage

**Review by ajlusk — APPROVED** [review]
Looks good to me!

A couple testing notes:
- Before step 7 the line added to simulate the error should be removed.
- NS Mayport is in the GBLOC CNNQ if you want to use an appropriate TOO for step 9.

HHG with simulated error:
Screenshot 2024-10-22 at 12 41 21 PM

PPM with PO box ZIP:
Screenshot 2024-10-22 at 1 05 42 PM

Very minor suggestion. The language here, "Please cancel and..." right next to the "Cancel Shipment" button is a bit misleading. Maybe "exit form" or something like that. Doesn't really matter since Prime Sim is internal only and it looks like you may have not introduced that language.
Screenshot 2024-10-22 at 1 07 49 PM

**Comment by antgmann** [comment]
> Very minor suggestion. The language here, "Please cancel and..." right next to the "Cancel Shipment" button is a bit misleading. Maybe "exit form" or something like that. Doesn't really matter since Prime Sim is internal only and it looks like you may have not introduced that language. 

I might change that, especially if I need to make more changes to get server test coverage to play nice. It has caused me to request to cancel shipments more than once. Thanks for the suggestion!

File: src/pages/PrimeUI/Shipment/PrimeUIShipmentUpdate.jsx [hunk 1]
```diff
@@ -44,6 +44,20 @@ const PrimeUIShipmentUpdate = ({ setFlashMessage }) => {
     navigate(generatePath(primeSimulatorRoutes.VIEW_MOVE_PATH, { moveCodeOrID }));
   };
 
+  const handleSetError = (error, invalidFieldsStr) => {
```
[the remark this answers is not in the document]
Comment (antgmann) [reply src/pages/PrimeUI/Shipment/PrimeUIShipmentUpdate.jsx:47]: Added a detailedMessage to hold the pre-formatted string and alter it based on if case to be put into the error message detail using the structure you provided. Cleaner as preformatted strings are preferred in our standards. Thanks for the suggestion, will be in the next push up.

**Review by MInthavongsay — APPROVED** [review]
Looks good!

**Comment by antgmann** [comment]
I believe I'm having client test coverage issues because of these lines, which may require actual error data instead of mocks? Looking into this. image

**Comment by antgmann** [comment]
I _think_ the client test coverage failing just because a comment already exists is only for server_test_coverage, but I just deleted the most recent client one just in case to see if it should be passing and just flagging because of the existing failure. 
Having trouble getting the coverage to hit those lines as apparently mocks can skip them. Still checking that out

File: pkg/handlers/primeapiv2/mto_shipment_test.go [hunk 2]
```diff
@@ -607,6 +607,42 @@ func (suite *HandlerSuite) TestCreateMTOShipmentHandler() {
...
+	suite.Run("POST failure - 500 - App Event Internal DTOD Server Error", func() {
+		// Under Test: CreateMTOShipmentHandler
+		// Setup:      Create a shipment with DTOD outage simulated or bad zip
+		// Expected:   500 Internal Server Error returned
+
+		handler, move := setupTestData(false)
+		req := httptest.NewRequest("POST", "/mto-shipments", nil)
+		handler.ShipmentCreator = &mockCreator
+
+		err := apperror.EventError{}
```
[the remark this answers is not in the document]
Comment (r-mettler) [reply pkg/handlers/primeapiv2/mto_shipment_test.go:619]: Maybe it has something to do with the case being flagged as not tracked.

**Comment by r-mettler** [comment]
Tests passed and code looks good.

**Review by r-mettler — APPROVED** [review]

File: pkg/handlers/primeapiv2/mto_shipment_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (antgmann) [reply pkg/handlers/primeapiv2/mto_shipment_test.go:619]: Whoever designed this didn't do it very nicely for those with Deuteranomaly

**Pull request merged**